### PR TITLE
LUN-230: Adjust tests for popover

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/routes/write-exercise/components/__tests__/MissingArticlePopover.spec.tsx [hunk 1]
```diff
@@ -5,21 +5,24 @@ import labels from '../../../../constants/labels.json'
...
   })
-
+  // Unfortunately jest renders the popover(modal) even if it's set to invisible
   it('should not show Popover when invisible', () => {
-    const { queryByText } = render(<MissingArticlePopover setIsPopoverVisible={jest.fn()} isVisible={false} />, {
+    const { getByTestId } = render(<MissingArticlePopover setIsPopoverVisible={jest.fn()} isVisible={false} />, {
       wrapper: wrapWithTheme
     })
-    expect(queryByText(labels.exercises.write.feedback.articleMissing)).toBeNull()
+    expect(getByTestId('popover').props.isVisible).toBeFalsy()
```
Comment (f1sh1918) [inline src/routes/write-exercise/components/__tests__/MissingArticlePopover.spec.tsx:26]: I know this isn't a good test, but seems like jest renders the modal in any case. you can check with `debug()`
Spent lots of time in it and couldn't find a better solution

File: src/routes/write-exercise/components/__tests__/MissingArticlePopover.spec.tsx [hunk 2]
```diff
@@ -5,21 +5,24 @@ import labels from '../../../../constants/labels.json'
...
+  default: 'Popover',
+  PopoverPlacement: 'top'
+}))
+
+describe('MissingArticlePopover', () => {
   it('should show Popover when visible', () => {
     const { getByText } = render(<MissingArticlePopover setIsPopoverVisible={jest.fn()} isVisible={true} />, {
       wrapper: wrapWithTheme
     })
     expect(getByText(labels.exercises.write.feedback.articleMissing)).toBeDefined()
```
Comment (SteffiStuffel) [inline src/routes/write-exercise/components/__tests__/MissingArticlePopover.spec.tsx:19]: but wouldn't his be always be defined, also if popover is invisible, as you wrote on the test below. So maybe we should check here also for the visible prop?

Comment (f1sh1918) [reply src/routes/write-exercise/components/__tests__/MissingArticlePopover.spec.tsx:13]: you're right. @Taggotty and me tried some stuff but didn't find a better solution so far

**Review by klinzo — APPROVED** [review]
I don't think we should spend too much time on this so fine for me :)

**Review by SteffiStuffel — APPROVED** [review]

File: src/routes/write-exercise/components/__tests__/InteractionSection.spec.tsx [hunk 3]
```diff
@@ -53,17 +63,15 @@ describe('InteractionSection', () => {
...
+    const { getByText, getByPlaceholderText, getByTestId } = renderInteractionSection(
       { ...document, result: null, numberOfTries: 0 },
       false
     )
     const inputField = getByPlaceholderText(labels.exercises.write.insertAnswer)
     fireEvent.changeText(inputField, 'Spachtel')
     const button = getByText(labels.exercises.write.checkInput)
     fireEvent.press(button)
-    await waitFor(() => expect(getByText(labels.exercises.write.feedback.articleMissing)).toBeDefined())
+    await waitFor(() => expect(getByTestId('popover').props.isVisible).toBeTruthy())
```
Comment (Taggotty) [inline src/routes/write-exercise/components/__tests__/InteractionSection.spec.tsx:75]: just as a hint. This is the same as `await findByTestId('...')`

**Review by Taggotty — APPROVED** [review]

**Pull request merged**